app/src/main/java/com/marcossalto/targetmvd/network/managers/LocationManager.kt

```diff
@@ -0,0 +1,53 @@
+package com.marcossalto.targetmvd.network.managers
+
+import android.content.Context
+import android.location.Location
+import com.google.android.gms.location.FusedLocationProviderClient
+import com.google.android.gms.location.LocationServices
+
+object LocationManager : ILocationManager {
+
+    private var fusedLocationProviderClient: FusedLocationProviderClient? = null
+
+    override fun saveLocation(lat: Double, lng: Double) {
+        UserLocation.latitude = lat
+        UserLocation.longitude = lng
+        UserLocation.locationState = LocationState.SUCCESS
```

Comment:
Use the scope function `.apply ` to assign values over the same object.
```suggestion
        UserLocation.apply {
            latitude = lat
            longitude = lng
            locationState = LocationState.SUCCESS
        }
```

Reply:
Thanks for your suggestion. I am going to apply it